realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -119,7 +119,7 @@ public String getClassName() {
      * @param className the new name for this class.
      * @throws IllegalArgumentException if className is {@code null} or an empty string, or its length exceeds 56
      * characters.
-     * @throws UnsupportedOperationException if this {@link RealmObjectSchema} is immutable.
+     * @throws UnsupportedOperationException if this {@link RealmObjectSchema} is immutable or from a synced Realm.
```

Comment:
Not entirely sure why we are using the word `immutable` here, since I don't think it is a concept we have introduced to users. We are referring to normal typed Realms (those you get from `Realm.getInstance`), but I suspect this isn't the PR to fix this in.

Reply:
Maybe it would be worth having different public types for the two types of schemas. All update methods of the _normal_ typed Realms' schema just throws `UnsupportedOperationException` and you only get an updatable scheme from the `DynamicRealm` (if I have gotten it right).

Reply:
Yes, it might be worth thinking about 👍